h/services/delete_user.py

```diff
@@ -19,38 +15,49 @@ def delete(self, user):
         """
         Deletes a user with all their group memberships and annotations.
 
-        Raises UserDeleteError when deletion fails with the appropriate error
-        message.
+        If a user owns groups with collaborators, meaning there are annotations
+        in the group that have been made by other users, the user is unassigned
+        as creator but the group persists.
         """
 
         created_groups = self.request.db.query(Group).filter(Group.creator == user)
-        if self._groups_have_anns_from_other_users(created_groups, user):
-            raise UserDeleteError(
-                "Other users have annotated in groups created by this user"
-            )
+        groups_to_unassign_creator = self._groups_that_have_collaborators(
+            created_groups, user
+        )
+        groups_to_delete = list(set(created_groups) - set(groups_to_unassign_creator))
 
         self._delete_annotations(user)
-        self._delete_groups(created_groups)
+        self._delete_groups(groups_to_delete)
+        self._unassign_groups_creator(groups_to_unassign_creator)
         self.request.db.delete(user)
 
-    def _groups_have_anns_from_other_users(self, groups, user):
+    def _groups_that_have_collaborators(self, groups, user):
         """
-        Return `True` if users other than `user` have annotated in `groups`.
+        Return list of groups that have annotations from other users.
+        :param groups: List of group objects to evaluate.
+        :type groups: h.models.Group
+        :param user: The user object and creator of the groups.
```

Comment:
Perhaps rename parameter to `creator` rather than having to state this in prose? Then it would suffice to omit the `:param user:` doc and just document the type.

Reply:
I think here I rather opt for consistency of using "user" as that's used everywhere else and keep this framed within the context of deleting a user.  Looking at just the context of this method though, I do think creator would be more explicit. I worry setting it as creator might lead to confusion or interpretation that "creator" is some how different from "user" in all the other usage locations.